src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/IdamServiceImpl.java

```diff
@@ -56,6 +63,11 @@ public IdamRolesInfo fetchUserById(String id) {
         try {
             Response response = idamClient.getUserById(id);
             result = buildIdamResponseResult(response);
+            HttpStatus httpStatus = HttpStatus.valueOf(response.status());
+            if (httpStatus.is5xxServerError()) {
+                result.setResponseStatusCode(HttpStatus.UNAUTHORIZED);
+                result.setStatusMessage(IdamStatusResolver.MISSING_TOKEN);
```

Comment:
All 5xx cannot be due to missing token. This message could mislead the users. Can you please send appropriate message?